.github/workflows/rust.yml

```diff
@@ -1,5 +1,40 @@
 name: Rust
 
+# On Rust, GitHub Actions, and caching
+# ===========
+# Here's a list of things to keep in mind if you find yourself maintaing this
+# CI:
+#
+# https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+#
+# - Always install and select the desired Rust toolchain *before* running
+#   `Swatinem/rust-cache`. This is because the active Rust toolchain is used as
+#   a cache key.
+# - When caching Rust compilation artifacts, keep in mind that different `cargo`
+#   commands will use different profiles
+#   (https://doc.rust-lang.org/cargo/reference/profiles.html). Learn what you
+#   can reuse between one job and another and don't assume two commands will
+#   just share caches without conflicts.
+# - Be extremely aware of cache thrashing a.k.a. churning. GitHub Actions' cache
+#   allows for 10GiB of data which is easily exceeded if not careful.
+#   Sometimes it's better not to cache than cache excessively.
+#   We might even want to disable caching writes for non-default branches
+#   altogether if cache churning is unacceptably high.
+# - Learn cache invalidation rules of `Swatinem/rust-cache` before making
+#   changes, e.g. what happens when `rustc --version` changes or `Cargo.lock`
+#   changes (or is missing).
+# - The jobs dependency tree is the way it is to accomodate for sharing caches,
+#   not necessarily because it makes logical sense to run one job after the
+#   other. This is due to the fact that we can't share caches between jobs that
+#   run in parallel.
+# - `sccache` is a good alternative to `Swatinem/rust-cache`, but it behaves
+#   poorly with GHA and often incurs into cache requests rate limits. We should
+#   probably explore `sccache` with a different backend.
```

Comment:
Can you please create an issue for this? Maybe next CI person can look into it.

Reply:
Tracked in https://github.com/Sovereign-Labs/sovereign-sdk/issues/584.